src/Data/Parameterized/Peano.hs

```diff
@@ -151,38 +160,81 @@ data PeanoView (n :: Peano) where
 
 -- | Test whether a number is Zero or Successor
 peanoView :: PeanoRepr n -> PeanoView n
+#ifdef UNSAFE_OPS
 peanoView (PeanoRepr i) =
-  if i == 0 then unsafeCoerce ZRepr else unsafeCoerce (SRepr (PeanoRepr (i-1)))
+  if i == 0
+  then unsafeCoerce ZRepr
+  else unsafeCoerce (SRepr (PeanoRepr (i-1)))
+#else
+peanoView = id
+#endif
 
 -- | convert the view back to the runtime representation
 viewRepr :: PeanoView n -> PeanoRepr n
+#ifdef UNSAFE_OPS
 viewRepr ZRepr     = PeanoRepr 0
 viewRepr (SRepr n) = PeanoRepr (peanoValue n + 1)
+#else
+viewRepr = id
+#endif
 
 ----------------------------------------------------------
 
 instance Hashable (PeanoRepr n) where
-  hashWithSalt i (PeanoRepr x) = hashWithSalt i x
+  hashWithSalt i x = hashWithSalt i (peanoValue x)
 
 instance Eq (PeanoRepr m) where
   _ == _ = True
 
 instance TestEquality PeanoRepr where
+#if UNSAFE_OPS
```

Comment:
The cabal file doesn't define a value for UNSAFE_OPS, so these should all be `#ifdef UNSAFE_OPS` instead of `#if UNSAFE_OPS`.

Reply:
Thanks. I've pushed an update.